pkg/pipeline/git_test.go

```diff
@@ -0,0 +1,349 @@
+package pipeline
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"testing"
+
+	synv1alpha1 "github.com/projectsyn/lieutenant-operator/pkg/apis/syn/v1alpha1"
+	"github.com/projectsyn/lieutenant-operator/pkg/git/manager"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestCreateOrUpdateGitRepo(t *testing.T) {
+	type args struct {
+		obj       *synv1alpha1.Cluster
+		template  *synv1alpha1.GitRepoTemplate
+		tenantRef v1.LocalObjectReference
+	}
+	tests := []struct {
```

Comment:
Please move this outside of the test function. And why is this an list and not a dict?

Does also apply to the other test cases within this file.

Reply:
This is a pattern I've often seen with golang tests, I'm not sure I understand your concern. Why should this be a dict?

Reply:
Using a list, requires you to have a name property within the struct. Using a map does give you that for free. On top of that, it ensures uniqueness of test names. I also find it easier to visually identify where a test case start and I immediately get an idea about its purpose.

I do not recall the URL but I have read an article that suggest using maps as a best practice.

Reply:
I see, thanks for the explanation. 

Since it is functionally equivalent, I don't feel strongly about changing the current solution however, personally.